### PR TITLE
Update charts to follow Helm RBAC best practices

### LINKED ISSUE #11755: RBAC and more customization in Helm Chart
Hello,

The Helm community has prepared [best practices for RBAC](https://helm.sh/docs/chart_best_practices/rbac/). 

In particular, it would be useful to make the following changes:
- The path to enable `rbac` should be `rbac.create` instead of `rbacEnabled`.
- It should be possible to disable the creation of each SA individually, but the service account should still be associated with the same resources so that manually-created RBAC resources created later that reference it will function correctly. 
-  It should be possible to configure its own name for each SA individually

Best regards,
Kamil Breguła

### PR DESCRIPTION
**Pull request by jaydesl**
Update the Helm charts to follow best practices around RBAC (as suggested in #11755) . Namely:
* `rbac.create` replaces `rbacEnabled`
* ServiceAccount creation can be enabled individually with `...serviceAccount.create`
* ServiceAccounts can be named with `...serviceAccount.name` - if not given, previous naming conventions are used
* WebServer SA creation is now conditional (previously it was created even if `rbacEnabled = False`)

closes: #11755 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by mik-laj** [comment]
I think it's worth adding more unit tests here to make sure everything works. Unfortunately, we are currently migrating to [new framework](https://github.com/apache/airflow/issues/11657) in which writing tests will be much easier and faster. This means that we have to wait to merge this change until we have a new framework.

**Comment by mik-laj** [comment]
Hello.

We have [a new test framework for Helm Chart](https://github.com/apache/airflow/blob/master/TESTING.rst#helm-unit-tests). Can you add unit tests?

Best regards,
Kamil Breguła

File: chart/values.schema.json [hunk 1]
```diff
@@ -425,6 +431,20 @@
                     "description": "Number of airflow celery workers in StatefulSet.",
                     "type": "integer"
                 },
+                "serviceAccount": {
```
Comment (mik-laj) [inline chart/values.schema.json:502]: Can you move `serviceAccountAnnotations` to this object?

**Comment by jaydesl** [comment]
Sure, I'll do both of these things today.

Jay

**Comment by mik-laj** [comment]
@jaydesl What is the status of this PR? Is it ready for review?

**Comment by DerekHeldtWerle** [comment]
@mik-laj @jaydesl, I've rebased this PR with master [here](https://github.com/DerekHeldtWerle/airflow/tree/fix/rbac) and added all of the additional requirements needed to get this chart deployed when working in environments where users have the [PodSecurityPolicy](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#podsecuritypolicy) admission controller enabled. 

When enabled, _every_ pod in the cluster must have a serviceaccount associated with it that is then binded to a psp. By having the option to set the serviceaccount for every potential pod allows teams to create a single service account mapped to a psp and set that for every pod. Long term, adding psp's at a per service level (e.g. airflow, flower, pgbouncer) would be the best path forward, but this is an initial step towards supporting that. 

I'm happy to open up a new PR that includes this PR's changes and more, but can wait until this PR is merged and add my changes after the fact if that is deemed more appropriate.

**Comment by mik-laj** [comment]
Related: https://github.com/apache/airflow/issues/13643

@FloChehab Could you please make a review and say if the changes presented are in line with your issue?

**Comment by jaydesl** [comment]
Hi @mik-laj sorry this fell off my radar. Let me know if you want me to rebase this or alternatively if its easier just go with the single PR from @DerekHeldtWerle (thanks btw!)

**Comment by FloChehab** [comment]
> Related: #13643
> 
> @FloChehab Could you please make a review and say if the changes presented are in line with your issue?

Hello @mik-laj, sorry for the response delay, I had other things on top of my mind lately.

The changes mentioned in the description of the PR looks like a step in the right direction, in particular in regards to #13643 & https://github.com/apache/airflow/issues/13643#issuecomment-774746562.

I would personally rebase this and handle the PSP related changes from @DerekHeldtWerle in a different PR (there still might some service accounts to add to this PR?).

Also, there might tests to add to make sure the annotations work properly (kind of like this: https://github.com/apache/airflow/pull/13615/files).

**Comment by DerekHeldtWerle** [comment]
@FloChehab 
> I would personally rebase this and handle the PSP related changes from @DerekHeldtWerle in a different PR (there still might some service accounts to add to this PR?).

My PR doesn't actually add any PSP related items, all it does it create all of the necessary service accounts that are required to be set when PSP's come into play as well as rebasing with master.

**Comment by FloChehab** [comment]
> @FloChehab
> 
> > I would personally rebase this and handle the PSP related changes from @DerekHeldtWerle in a different PR (there still might some service accounts to add to this PR?).
> 
> My PR doesn't actually add any PSP related items, all it does it create all of the necessary service accounts that are required to be set when PSP's come into play as well as rebasing with master.

Sorry I think I saw some things related to uids and I thought there might be some more complex things related to psp in your PR.

I don't really have an opinion on what is best to do. I am sure we will end up in the right spot some way or another.

**Comment by DerekHeldtWerle** [comment]
@jaydesl @FloChehab I went ahead and rebased with master again and opened up a new PR [here](https://github.com/apache/airflow/pull/14152) to keep this moving along. Let me know if there's anything on the PR that needs to be addressed and I'll get to it as soon as I can 😄

**Pull request closed without merging**